### PR TITLE
ipq806x: move to 6.1 and use nvmem fixed-layout

### PR DESCRIPTION
**Pull request by Ansuel**
Move to kernel 6.1 by default in preparation for DSA introduction.

Drop mac-address-increment from DT and move to new nvmem fixed-layout
implementation.

### CODE REVIEW

File: target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr42.dts [hunk 1]
```diff
@@ -87,9 +87,17 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
```
Comment (Djfe) [inline target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr42.dts:89]: This is missing `compatible = "nvmem-cells";` in Line 83. I'm not sure whether it's possible to combine it with `compatible = "atmel,24c64";`

File: target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr52.dts [hunk 2]
```diff
@@ -145,9 +144,17 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
```
Comment (Djfe) [inline target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr52.dts:146]: Same as above

**Review by Djfe — COMMENTED** [review]
I looked over all files, that your nvmem commit changed. Only found the issue with "compatible", which might be no issue, since it was already in place before this.

Do you also want to (eventually) migrate devices, that use nvmem but have enough macs so that they never required mac-increment, to the new nvmem-layout?
https://github.com/Ansuel/openwrt/blob/ipq806x-layout/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8062-wg2600hp3.dts#L467

**Comment by Ansuel** [comment]
the cell binding is needed only for mtd. The eeprom driver expose cell on
their own.

The idea is that we have enough support to drop 02_network entirely

Il Mar 18 Lug 2023, 12:19 Felix Baumann ***@***.***> ha
scritto:

**Comment by castiel652** [comment]
Got this 
`Error: /home/castiel652/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/linux-ipq806x_generic/linux-6.1.35/arch/arm/boot/dts/qcom-ipq8068-mr42.dts:101.2-3 syntax error;`

File: target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr42.dts [hunk 3]
```diff
@@ -87,9 +87,17 @@
...
+			compatible = "fixed-layout";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mac_address: mac-address@66 {
+				compatible = "mac-base";
+				reg = <0x66 0x6>;
+				#nvmem-cell-cells = <1>;
+			};
+		}
```
Comment (Djfe) [inline target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr42.dts:100]: @castiel652 just discovered a missing semicolon

File: target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr52.dts [hunk 4]
```diff
@@ -145,9 +144,17 @@
...
+			compatible = "fixed-layout";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mac_address: mac-address@66 {
+				compatible = "mac-base";
+				reg = <0x66 0x6>;
+				#nvmem-cell-cells = <1>;
+			};
+		}
```
Comment (Djfe) [inline target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr52.dts:157]: same as above, semicolon missing

**Comment by Djfe** [comment]
Good catch, copy&paste mistake.
atleast 4 nvmem-layout's are missing a semicolon on their closing bracket

**Comment by rmilecki** [comment]
Thanks a lot for that great `"mac-base"` effort!

**Pull request closed without merging**